machine/v1beta1/types_awsprovider.go

```diff
@@ -44,6 +44,12 @@ type AWSMachineProviderConfig struct {
 	// it should use the default of its subnet.
 	// +optional
 	PublicIP *bool `json:"publicIp,omitempty"`
+	// NetworkInterfaceType specifies the type of network interface to be used for the primary
+	// network interface.
+	// Valid values are "interface", "efa", and omitted, which means no opinion and the platform
+	// chooses a good default which may change over time.
+	// +optional
```

Comment:
I know we don't use openapi validations, but can we have an enum here, just to be consistent with what will be added in upstream CAPI?